### PR TITLE
Navigation style redesign

### PR DESCRIPTION
**Pull request by Christian862**
## Description
These changes introduce new styling to the navigation list. This styling is for the overall project of navigation redesign but also includes backwards compatibility via `NavViewStyle` input option to the nav components. With these changes, if you are consuming NavigationList or NavigationItem and do not specify the previously mentioned input option, the styles will remain the same as always. With the input option, the components will take on a 'dark' style.

### Testing
Please describe the tests that you ran to verify your changes. Please summarize what did you test and what needs to be tested e.g. deployed and tested helm chart locally. 

### Checklist:
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### Documentation
Make sure that you have documented corresponding changes in this repository or [hypertrace docs repo](https://github.com/hypertrace/hypertrace-docs-website) if required.

### CODE REVIEW

File: projects/components/src/navigation/navigation.config.ts [hunk 1]
```diff
@@ -47,3 +47,14 @@ export const enum NavItemType {
   Divider = 'divider',
   Footer = 'footer'
 }
+
+export interface NavItemGroup {
```
Comment (anandtiwary) [inline projects/components/src/navigation/navigation.config.ts:51]: We already have a header link that kind of forms a group with the following links. Let me cherry-pick this pR and play around with it

Comment (Christian862) [reply projects/components/src/navigation/navigation.config.ts:51]: conversation resolved in slack

**Comment by anandtiwary** [comment]
Looks fine. @Christian862 Have you tested this with both flag enabled and disabled?

**Review by anandtiwary — APPROVED** [review]

**Comment by anandtiwary** [comment]
Going to trying closing and reopening the PR to trigger the build

**Pull request merged**